src/utils/temp.ts

```diff
@@ -0,0 +1,122 @@
+import * as uuid from 'uuid';
+import neo4j, { Driver, QueryResult, Session } from 'neo4j-driver';
+
+const getCurrentTimestamp = () => {
+  return Math.floor(new Date().getTime() / 1000);
+};
+
+const filterConsoleDatabasesFromResult = (result: QueryResult<any>) => {
+  return result.records.filter(
+    (record) => record.get('name').indexOf('temp--') === 0,
+  );
+};
+
+const deleteDatabaseUserAndRole = async (
+  session: Session,
+  database: string,
+) => {
+  try {
+    await session.run(`STOP DATABASE ${database};`);
+  } catch (error) {
+    console.error(error);
+  }
+  try {
+    await session.run(`DROP DATABASE ${database};`);
+  } catch (error) {
+    console.error(error);
+  }
+  try {
+    await session.run(`DROP USER ${database};`);
+  } catch (error) {
+    console.error(error);
+  }
+  try {
+    await session.run(`DROP ROLE ${database};`);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+export const createTempDatabase = async (driver: Driver) => {
+  const sessionId = uuid.v4().replace(/-/g, '');
+  const currentTimestamp = getCurrentTimestamp();
+  const database = `temp--${sessionId}${currentTimestamp}`;
+
+  const session = driver.session({
+    database: 'system',
+    defaultAccessMode: neo4j.session.WRITE,
+  });
+
+  try {
+    await session.run(`CREATE DATABASE ${database} WAIT;`);
+    await session.run(
+      `CREATE USER ${database} SET PASSWORD '${database}' SET PASSWORD CHANGE NOT REQUIRED;`,
+    );
+    await session.run(`CREATE ROLE ${database};`);
+    await session.run(`GRANT ROLE ${database} TO ${database};`);
+    await session.run(`GRANT ALL ON DATABASE ${database} TO ${database};`);
+    await session.run(`GRANT ACCESS ON DATABASE ${database} TO ${database};`);
+    await session.run(`GRANT READ {*} ON GRAPH ${database} TO ${database};`);
+    await session.run(`GRANT TRAVERSE ON GRAPH ${database} TO ${database};`);
+    await session.run(`GRANT WRITE ON GRAPH ${database} TO ${database};`);
+  } catch (error) {
+    console.error(error);
+  } finally {
+    await session.close();
+  }
+
+  return database;
+};
+
+export const clearTempDatabase = async (driver: Driver, database: string) => {
+  const session = driver.session({ database: 'system' });
+  try {
+    await deleteDatabaseUserAndRole(session, database);
+  } catch (error) {
+    console.error(error);
+  } finally {
+    await session.close();
+  }
+};
+
+export const clearTempDatabasesOlderThan = async (
+  driver: Driver,
+  seconds: number,
+) => {
+  const session = driver.session({ database: 'system' });
+  const result = await session.run('SHOW DATABASES');
+  const shouldExpireAt = getCurrentTimestamp() - seconds;
+  try {
+    const records = filterConsoleDatabasesFromResult(result);
+    console.log('Databases found: ' + records.length);
+    for (const record of records) {
+      const database = record.get('name');
+      const dbTimestamp = parseInt(database.slice(39), 10);
```

Comment:
I'd like if you could add a comment to indicate what the format is - so it's clear why we're slicing it at 39.

Reply:
Gotcha

Reply:
Done